packages/order/integration-tests/__tests__/create-order.ts

```diff
@@ -0,0 +1,210 @@
+              adjustments: [
+                {
+                  code: "VIP_10",
+                  amount: 1,
+                  description: "VIP discount",
+                  promotion_id: "prom_123",
+                },
+              ],
+            },
+          ],
+          currency_code: "usd",
+          customer_id: "joe",
+        } as CreateOrderDTO)
+
+        const getOrder = await service.retrieve(order.id, {
+          select: [
+            "id",
+            "version",
+            "items.id",
+            "items.version",
+            "items.item.id",
+            "items.item.tax_lines.id",
+            "items.item.adjustments.id",
+            "shipping_address.id",
+            "billing_address.id",
+            "shipping_methods.id",
+            "shipping_methods.tax_lines.id",
+            "shipping_methods.adjustments.id",
+          ],
+          relations: [
+            "shipping_address",
+            "billing_address",
+            "items",
+            "items.item",
+            "items.item.tax_lines",
+            "items.item.adjustments",
+            "shipping_methods",
+            "shipping_methods.tax_lines",
+            "shipping_methods.adjustments",
+          ],
+        })
+
+        expect(getOrder).toEqual(
+          expect.objectContaining({
+            id: expect.stringContaining("order_"),
+            version: 1,
+            shipping_address: {
+              id: expect.stringContaining("ordaddr_"),
+            },
+            billing_address: {
+              id: expect.stringContaining("ordaddr_"),
+            },
+            items: [
+              expect.objectContaining({
+                id: expect.stringContaining("orddetail_"),
+                version: 1,
+                item: expect.objectContaining({
+                  id: expect.stringContaining("ordli_"),
+                  tax_lines: [
+                    expect.objectContaining({
+                      id: expect.stringContaining("ordlitxl_"),
+                    }),
+                  ],
+                  adjustments: [
+                    expect.objectContaining({
+                      id: expect.stringContaining("ordliadj_"),
+                    }),
+                  ],
+                }),
+              }),
+              expect.objectContaining({
+                id: expect.stringContaining("orddetail_"),
+                version: 1,
+                item: expect.objectContaining({
+                  id: expect.stringContaining("ordli_"),
+                  tax_lines: [],
+                  adjustments: [],
+                }),
+              }),
+              expect.objectContaining({
+                id: expect.stringContaining("orddetail_"),
+                version: 1,
+                item: expect.objectContaining({
+                  id: expect.stringContaining("ordli_"),
+                  tax_lines: [],
+                  adjustments: [],
+                }),
+              }),
+            ],
```

Comment:
I really don't like this structure - specifically:
- `items.item` nesting is not pretty
- it doesn't follow the same pattern as cart
- it's not intuitive what the point of "orddetail_" is, which will confuse an external developer

Two options I'd be fine with:

1. invert how orddetail_ is stored - i.e. something like this

```
item: [
  {
    id: "ordli_12313",
    tax_lines: [..],
    adjustments: [...],
    ...,
    item_detail: {
      id: "orddetail_123132",
      version: 1
    }
  }
]
```

2. Create an `item_details` field on the order item that can store in the current format. But then keep `items` for pure items.

```
item: [
  {
    id: "ordli_12313",
    tax_lines: [..],
    adjustments: [...],
    ...,
  }
],
item_details: [
  {
    id: "orddetail_123132",
    version: 1,
    item: { ... }
  }
]
```

Reply:
Also, what is the idea with order detail?

Reply:
Order details will handle versioning and all the totals of the item for that specific version.
I will transform it before returning, where do you think these quantities like `return_requested_quantity`, `shipped_quantity` should be placed?
also, there is the metadata inside these details where custom quantities/values could be placed.

can it be something like this:
```
{
  order: {
    items: [
      {
        id: "ordli_12313",
        tax_lines: [...],
        adjustments: [...],
        detail: {
          version: 1,
          shipped_quantity: 1,
          ...,
        }
      }
    ]
  }
}
```

Reply:
I will rename it to `order_items` internally to make more sense.

Reply:
Think this is perfect 👍 

> can it be something like this?
> ```
> {
>  order: {
>    items: [
>      {
>        id: "ordli_12313",
>        tax_lines: [...],
>        adjustments: [...],
>        detail: {
>          version: 1,
>          shipped_quantity: 1,
>          ...,
>        }
>      }
>    ]
>  }
>}
>```